src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -355,6 +356,9 @@ internal static string GetFilePathPrefixForGenerator(ISourceGenerator generator)
 
         private static (ImmutableArray<ISourceGenerator>, ImmutableArray<IIncrementalGenerator>) GetIncrementalGenerators(ImmutableArray<ISourceGenerator> generators, string sourceExtension)
         {
+            var builder = generators.ToBuilder();
+            builder.SortByDependency();
+            generators = builder.ToImmutable();
```

Comment:
this PR does not pretend to load dependent generators, but to ensure that they run before or after generators if they are present in the pipeline. See PR test. Perhaps it would be appropriate to report a warning.

Reply:
@workgroupengineering If all the generators will still see the original source code, then this wouldn't address the associated issues anyway, right? I don't think it matters what order they run in if they can't see output from another, so I don't understand the purpose of this change.